### PR TITLE
PR 7

### PR DESCRIPTION
**Pull request by jokano12**
Add Category to an Organization
Add Customer Request
Update Customner Request
Delete a Category for an Organization
Get a List of All Notifications
List all Exhibits
Change the Password of a User
Get all states by country
Get Membership Stats by City
Get Membership Stats by State
List all Countries
Send Password Reset Link
Update Task by Task Number
Add Activity to Sales Opportunity
Add Note to Sales Opportunity

RequestSaveData
RequestUpdateData
ExhibitData
PasswordData
StateProvinceData
CityData
MembershipStatsStateData
CountryData
BaseNoteData

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by lordlydon** [comment]
@jokano12, can you check the conflicts?

**Comment by jokano12** [comment]
@LordLydon  done

**Comment by lordlydon** [comment]
Hi @jokano12, can you check your PR again? the new actions look good but it seems like you removed the new webhooks that were added

**Comment by jokano12** [comment]
@LordLydon Cant you make your own PR to this repository from your own repository with the webhooks after you fetch the new actions?

**Review by lordlydon — CHANGES_REQUESTED** [review]
For some reason I can't add comments on the changes itself (maybe because there are so many)

### For PR-8:

There are some properties in definitions with whitespaces in their name and this is causing an issue when running the flows where the values are not mapped properly.
This seem to only happen if the whitespaces appear in the name of the definition for a response, for whitespaces in the parameters in body, query and url seem to work.

For example for `Get Rooms` the response schema `RoomData` has the property `Max Capacity` but it's value can't be used in a flow as it isn't mapped because the API sends `maxCapacity` instead.

Also, there are some actions that are missing:
`Get Customer Notification Count`
`Get Individual Custom Field Values`  - There is only the update/post but no get

The results for the action `Add or Update a List of Custom Fields Per Organization` should be an array of `CustomFieldResultData`, in the definition it's currently a single object (Be aware that there is an error in the swagger definition, so the help page for this one shows that it returns a DataCollection but it actually returns an array).

There is an error in the swagger definition for the result of the `Get Presentation Topic by Session` endpoint, it actually returns a DataCollection of `PresentationTopicData`, can you please update the action?

In line 18356, for `SpeakerData.Sessions.RelatedProducts` there is a typo in the property `IsRequire` it should be `IsRequired` 

### PR-7:

The actions from PR-7 look good.

**Comment by jokano12** [comment]
@LordLydon  I made the following changes: 

Changed "IsRequired"

Made Add or Update a List of Custom Fields Per Organization an Array

Added action Get Individual Custom Field Values

Removed Whitespaces in the definitions that I could find, as per your suggestion.

Not Changed:

Get Customer Notification Count was one of the actions listed to be removed, do you want it readded?

Get Presentation Topic by Session endpoint to DataCollection - Im not certain, what I need to change, it appears to me that its formatted how the other dataCollections are. What exactly do I need to add or remove here?

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 1]
```diff
@@ -1732,6 +1732,7 @@
           "200": {
             "description": "default",
             "schema": {
+              "type": "array",
               "$ref": "#/definitions/CustomFieldResultData"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: The $ref should be inside items, like this:
```json
"schema": {
  "type": "array",
  "items": {
    "$ref": "#/definitions/CustomFieldResultData"
  }
}
```

**Comment by lordlydon** [comment]
Hi @jokano12,

> Not Changed:
> 
> Get Customer Notification Count was one of the actions listed to be removed, do you want it readded?

No, I just wanted to check if those actions were missing for a reason because you mentioned them in the list of actions for the PR but I didn't see it in the definitions.

> Get Presentation Topic by Session endpoint to DataCollection - Im not certain, what I need to change, it appears to me that its formatted how the other dataCollections are. What exactly do I need to add or remove here?


For this one, the problem is not the dataCollection format, but the type of object in the dataList. There was an error in the OpenAPI definition for this endpoint that makes it display the wrong type in the documentation.

In this case, the API returns a dataCollection but of **`PresentationTopicData`** (The same one sent as the body in `Add Presentation Topic to an Event Session`) not `SpeakerData` so we need to update the $ref in that response to use the correct definition.

**Review by lordlydon — CHANGES_REQUESTED** [review]

**Comment by jokano12** [comment]
@LordLydon  Just to be clear about the Get Presentation Topic by Session. You want me to change the $ref to PresentationTopicData from SpeakerData?

**Comment by lordlydon** [comment]
That’s correct
________________________________
De: jokano12 ***@***.***>
Enviado: Tuesday, January 18, 2022 4:56:23 PM
Para: RjImpx/PowerPlatformConnectors ***@***.***>
Cc: William Llanos ***@***.***>; Mention ***@***.***>
Asunto: Re: [RjImpx/PowerPlatformConnectors] PR 7 (PR #28)


@LordLydon<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FLordLydon&data=04%7C01%7C%7Cb08a75b037b1410550db08d9dacd5e96%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637781397869278325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=LUqzTNiETFaLPl1ZhsO%2BMEE%2BR9AsmNiRh1G6dPzZ4MY%3D&reserved=0> Just to be clear about the Get Presentation Topic by Session. You want me to change the $ref to PresentationTopicData from SpeakerData?

—
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FRjImpx%2FPowerPlatformConnectors%2Fpull%2F28%23issuecomment-1015870073&data=04%7C01%7C%7Cb08a75b037b1410550db08d9dacd5e96%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637781397869278325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=36BoWkeOnR1k%2BjsKWcOoRL2jieaDzGA3W%2BfD1fNA9%2FI%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACJSFRLCNDTBUFKQLBSKVJ3UWXOYPANCNFSM5KTBXEEQ&data=04%7C01%7C%7Cb08a75b037b1410550db08d9dacd5e96%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637781397869278325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2FxPTN5VkjUU9E7dVVhUqq1YElaNlUj9fmQAnjqRU8bw%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7Cb08a75b037b1410550db08d9dacd5e96%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637781397869278325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=eBtz%2B%2BaNqC4vHl2O088SxwhXgUSeL2Rz2gpOQSf1EoI%3D&reserved=0> or Android<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7Cb08a75b037b1410550db08d9dacd5e96%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637781397869278325%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=0ZyeP5kKPGSH%2FgvCf3qe72PBqRwZiSvzuzYI7Yms6Kk%3D&reserved=0>.

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 2]
```diff
@@ -1733,8 +1733,9 @@
             "description": "default",
             "schema": {
               "type": "array",
-              "$ref": "#/definitions/CustomFieldResultData"
-            }
+              "items": {
+                "$ref": "#/definitions/CustomFieldResultData"
+              }
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Missing a closing bracket `}`

**Review by lordlydon — CHANGES_REQUESTED** [review]

**Review by lordlydon — APPROVED** [review]
Looks good.

**Pull request merged**